standards/uipath_standards.md

```diff
@@ -1,88 +1,134 @@
+
+#### Image based automation
+Keep this as a last resort - image based activities are slow and unstable.
+
+#### Nesting activities
+E.g. an if within an if within … Nesting activities within each other makes the code difficult to read. Try to change your approach to reduce the level of nesting. E.g. use a switch case instead of a nest of ifs.
+
+#### Spaghetti code
+Spaghetti code refers to a difficult to follow coding style. Your code should be easy to follow in a flowchart. E.g. Having crossed lines in a flow chart will make it difficult to follow. E.g. Splitting a function written in several different invokes.
+
+#### Storing passwords/sensitive information in Orchestrator
+Some of our stakeholders do not authorise the use of cloud services. Please check with stakeholders for the preferred method of handling data.
+
+#### Leaving developer code in final product main files
+It is acceptable to leave separate testing xaml’s however avoid leaving commented out code in the main code or temporary write-lines etc…
+
+#### Exception swallowing
+If you use a try catch ensure exceptions are logged as a minimum because an empty catch makes it difficult to pinpoint the root cause during debugging.
+
+#### Complicated/complex coding techniques
+Avoid using complex coding techniques that could be done with simpler methods. If complex code is required please include an explanation within the annotation of the code. E.g. Using invoke code with vb.net instead of using predefined activities.
+
+#### Complicated variables
+E.g. Use a datatable/json object instead of dictionary within a dictionary variable.
+
+#### Lasagne code
+It is good practice to break up a flow into smaller flows however if this is done frequently the code becomes too layered. Try to keep how deep invokes within invoke layers reach. In addition don’t use one xaml file for everything! Avoid the use of single activity invokes or single activity components.
+
+#### High number of variables
+Try to reduce the number of variables used. It is difficult to track information being transferred during debugging if it’s passed from variable to variable. Try to keep arguments and variables that connect to each other in different xamls similarly named. You could make use of global variables if possible. Add into config variable if you need to use variable everywhere.
+
+#### Using delay activity
+Try to keep the delay activity as a last resort. You should be able to use find element as a wait which will stop waiting as soon as confirmation is received e.g. click followed by a find element. Consider also reducing the delays in the activities to speed up your code.
+
+## Status
+
+This standard was formally adopted on 23 July 2024.
+
+## Significant changes
+
+Re-write of UiPath best practices based upon the ways of working implemented 2021 onwards.
```

Comment:
Change to

> Original version adopted October 2020.
> Re-write July 2024 based on the revised ways of working implemented from 2021 onwards .

Reply:
It's just missing when the original version was adopted - it's good to have this on here so that people can easily see they may have been working against and older version of the standard